client/src/components/Book/BookForm.js

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect } from "react";
+import { Redirect } from "react-router-dom";
+import ErrorList from "../ErrorList.js";
+import translateServerErrors from "../../services/translateServerErrors.js";
+
+const BookForm = (props) => {
+  const [bookRecord, setBookRecord] = useState({
+    title: "",
+    author: "",
+    bookList: "",
+  });
+  const [bookLists, setBookLists] = useState([]);
+  const [errors, setErrors] = useState([]);
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+
+  const fetchBookLists = async () => {
+    try {
+      const response = await fetch(`/api/v1/bookLists`);
+
+      if (!response.ok) {
+        throw new Error(`${response.status} (${response.statusText})`);
+      }
+
+      const body = await response.json();
+      setBookLists([{ name: "", id: "" }, ...body.bookLists]);
+    } catch (err) {
+      console.error(`Error in fetch: ${err.message}`);
+    }
+  };
+
+  const addBook = async (bookPayload) => {
+    try {
+      const response = await fetch(`/api/v1/books`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json",
+        }),
+        body: JSON.stringify(bookPayload),
+      });
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json();
+          const newErrors = translateServerErrors(body.errors);
+          setErrors(newErrors);
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`;
+          const error = new Error(errorMessage);
+          throw error;
+        }
+      } else {
+        setErrors([]);
+        setShouldRedirect(true);
+      }
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`);
+    }
+  };
+
+  useEffect(() => {
+    fetchBookLists();
+    if (shouldRedirect) {
+      return <Redirect to="/books" />;
+    }
```

Comment:
this if statement will only be evaluated on initial render, and we never want to render something anyways from a useEfffect. Move this logic outside of your useEffect, ideally towards the rest of your JSX further down in the file. It doesnt need to be in a function